### PR TITLE
Add LSP hint in invoice.

### PR DESCRIPTION
**Pull request by roeierez**
This PR adapts the `AddInvoice` endpoint to support **Pay to Open** flow.
Specifically in case the capacity is not enough the LSP info is used to populate routing hints that will be used to route the the payment in case there is no channel yet.
When the LSP intercepts the htlc, an unconfirmed channel is opened which is used to route the htlc to the destination.

### CODE REVIEW

File: account/invoice.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+)
+
+const (
+	fakeShortChannelID = 402099585968178542
+)
+
+func (a *Service) generateInvoiceWithNewAmount(payReq string, newAmount int64) (string, error) {
+	invoice, err := zpay32.Decode(payReq, a.activeParams)
+	if err != nil {
+		return "", fmt.Errorf("zpay32.Decode() error: %v", err)
```
Comment (yaslama) [inline account/invoice.go:25]: Use %w instead of %v in order to preserve the original error

File: account/payments.go [hunk 2]
```diff
@@ -193,57 +194,84 @@ func (a *Service) sendPayment(paymentHash string, sendRequest *lnrpc.SendRequest
...
+
+		// Calculate the channel fee.
+		amountForFeeMsat := float64(amountMsat - (lspInfo.ChannelFeeStartAmount * 1000))
+		channelFeeMsat := amountForFeeMsat * float64(lspInfo.ChannelFeeRate)
+		if channelFeeMsat < 0 {
+			channelFeeMsat = 0
 		}
-		a.log.Infof("adding routing hint = %v", h.RemotePubkey)
+
+		fakeChanID := &lnwire.ShortChannelID{BlockHeight: 1, TxIndex: 0, TxPosition: 0}
```
Comment (yaslama) [inline account/payments.go:241]: I suggest perhaps to use BlockHeight: 1 >> 17 to be consistent with the zeroconf fake chanID

File: account/invoice.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/lightningnetwork/lnd/lnrpc/signrpc"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/zpay32"
+)
+
+const (
+	fakeShortChannelID = 402099585968178542
```
Comment (yaslama) [inline account/invoice.go:19]: It's used?

File: account/payments.go (hunk 2 above)
Comment (roeierez) [reply account/payments.go:241]: My thought was actually to avoid this range so I won't need to check for conflicts with existing channels...

**Pull request merged**